generators/earlgrey

```diff
@@ -36,14 +37,31 @@ sources = ''
 incdirs = ''
 incs = []
 
+subprocess.call(
+    [
+        "fusesoc", "--cores-root", earlgrey_path, "sim",
+        "lowrisc:systems:top_earlgrey:0.1"
+    ],
+    stdout=subprocess.DEVNULL,
+    stderr=subprocess.DEVNULL)
+
 earlgrey_conf_path = os.path.abspath(
-    os.path.join(conf_dir, "generators", "earlgrey"))
+    os.path.join("build", "lowrisc_systems_top_earlgrey_0.1", "sim-icarus"))
 
-with open(os.path.join(earlgrey_conf_path, 'earlgrey.src'), "r") as f:
+with open(os.path.join(earlgrey_conf_path,
+                       'lowrisc_systems_top_earlgrey_0.1.scr'), "r") as f:
     for line in f:
         line = line.strip()
-        incs.append(os.path.dirname(os.path.join(earlgrey_path, line)))
-        sources += os.path.join(earlgrey_path, line) + ' '
+        if "+incdir+" in line:
+            line = line.replace('+incdir+', '')
+            incs.append(
+                os.path.dirname(os.path.join(earlgrey_conf_path, line)))
+            continue
+        elif "+define+" in line:
+            continue
```

Comment:
I really think we need a mechanism to use those defines if possible. These should be somehow passed to the runners (via another `:marker:`) and selected runners should be able to use it. I think it is ok if we only add it to verilator at first.